src/components/shared/formItems/ProjectHandle.tsx

```diff
@@ -54,21 +57,27 @@ export default function ProjectHandle({
     }
   }, [inputContents])
 
-  // InputContents pattern allows checking if handle exists while typing
+  // inputContents pattern allows checking if handle exists while typing
   const idForHandle = useContractReader<BigNumber>({
     contract: ContractName.Projects,
     functionName: 'projectFor',
     args: handle && requireState ? [handle] : null,
     callback: useCallback(
-      returnValue === 'id'
-        ? id => onValueChange(id?.toHexString() ?? '0x00')
-        : () => null,
-      [returnValue],
+      id => {
+        setHandleLoading(false)
+        if (returnValue === 'id' && onValueChange) {
+          onValueChange(id?.toHexString() ?? '0x00')
+        }
+      },
+      [returnValue, onValueChange],
     ),
   })
 
-  const handleExists = idForHandle?.gt(0)
+  const handleExists = Boolean(idForHandle?.gt(0))
```

Comment:
what's the purpose here? does `Boolean(x)` convert all falsy values to `false`?

Reply:
> does Boolean(x) convert all falsy values to false?

Exactly 👍